### PR TITLE
Update pattern

### PR DESCRIPTION
**Pull request by idleberg**

### CODE REVIEW

**Comment by idleberg** [comment]
Thanks Ivan and sorry for flooding you, should've tested this locally!

**Comment by isagalaev** [comment]
No problem at all :-) Let Travis work, it's a robot it doesn't care :-)

File: src/languages/nsis.js [hunk 1]
```diff
@@ -8,26 +8,26 @@ Website: http://github.com/idleberg
...
-    begin: '\\$+{[a-zA-Z0-9_]+}'
+    begin: /\$+{[\w\.:-]+}/
   };
 
   var VARIABLES = {
     // $variables
     className: 'variable',
-    begin: '\\$+[a-zA-Z0-9_]+',
-    illegal: '\\(\\){}'
+    begin: /\$+[\w]+/,
```
Comment (isagalaev) [inline src/languages/nsis.js]: You don't need square brackets for a single choice, so: `/\$+\w+/`

File: src/languages/nsis.js [hunk 2]
```diff
@@ -8,26 +8,26 @@ Website: http://github.com/idleberg
...
+    illegal: /\(\){}/
   };
 
   var LANGUAGES = {
     // $(language_strings)
     className: 'variable',
-    begin: '\\$+\\([a-zA-Z0-9_]+\\)'
+    begin: /\$+{[[\w\.:-]+]+}/
```
Comment (isagalaev) [inline src/languages/nsis.js]: One of the nested pair of `[..]+` is excessive and may even lead to exponential performance degradation if not optimized away by a regex engine. Please drop one!

**Comment by isagalaev** [comment]
Merged! Thanks :-)

**Pull request merged**